### PR TITLE
ID Card Console tweaks

### PR DESCRIPTION
**Pull request by CinnamonSnowball**
## What Does This PR Do
- Fixes department management consoles denying access to department head IDs with custom titles (e.g. `Acting CMO`)
- Fixes ID consoles failing to highlight rank of inserted ID if it has a custom title (`Nurse` wouldn't highlight `Medical Doctor`)
- Unifies full identification and department management console UIs
- Moves tabs below the authentication section (since it doesn't care about which tab you're on)
- Fixes being able to reassign a terminated ID on a departmental console despite them explicitly not allowing reassignment of demoted IDs
- Fixes demoted IDs not blocking the option to re-demote them

## Why It's Good For The Game
Acting heads actually able to use their management consoles is definitely nice, and being able to see what underlying rank does a custom-title ID have is nifty as well.

## Testing
See above; tested various combinations of IDs and ID consoles to check that no clearance level has lost or gained any privileges

## Changelog
:cl:
tweak: Tweaked ID computer UI slightly to be more intuitive
fix: Fixed departmental management consoles denying access to IDs with command ranks but non-standard titles
fix: Fixed ID computers not highlighting current rank of inserted card properly
fix: Fixed being able to reassign a terminated ID on a departmental management console
fix: Fixed being able to (attempt to) re-demote a demoted ID
/:cl:

### CODE REVIEW

**Review by S34NW — APPROVED** [review]
It looks fine to me, but I'm not a very good TGUI nerd. Perhaps @Sirryan2002 can take a peek? Anyway, please deconflict.

**Review by Sirryan2002 — APPROVED** [review]
the TGUI for this `.js` file is pretty awful, but its outside of the scope for this PR, the changes made in this PR are fine however.

File: code/controllers/subsystem/jobs.dm [hunk 1]
```diff
@@ -634,7 +634,7 @@ SUBSYSTEM_DEF(jobs)
 	if(tgtcard)
 		var/mob/M = tgtcard.getPlayer()
 		for(var/datum/job/job in occupations)
-			if(tgtcard.assignment && tgtcard.assignment == job.title)
+			if(tgtcard.rank && tgtcard.rank == job.title)
```
Comment (AffectedArc07) [inline code/controllers/subsystem/jobs.dm:637]:
```suggestion
			if(tgtcard?.rank == job.title)
```

File: code/game/machinery/computer/card.dm [hunk 2]
```diff
@@ -312,7 +309,8 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
 	data["mode"] = mode
 	data["modify_name"] = modify ? modify.name : FALSE
 	data["modify_owner"] = modify && modify.registered_name ? modify.registered_name : "-----"
-	data["modify_rank"] = get_target_rank()
+	data["modify_rank"] = modify && modify.rank ? modify.rank : FALSE
+	data["modify_assignment"] = modify && modify.assignment ? modify.assignment : "Unassigned"
```
Comment (AffectedArc07) [inline code/game/machinery/computer/card.dm:313]:
```suggestion
	data["modify_rank"] = modify?.rank ? modify.rank : FALSE
	data["modify_assignment"] = modify?.assignment ? modify.assignment : "Unassigned"
```

**Review by AffectedArc07 — DISMISSED** [review]
Minor nitpicks

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**